src/_playground/Routes.js

```diff
@@ -270,59 +271,97 @@ const routes = [
     }
 ];
 
-export const Routes = () => {
-    let sectionRoutes;
-    const groupedRoutes = groupArray(routes, 'section');
+export class Routes extends Component {
+    constructor(props) {
+        super(props);
+        this.state = {
+            query: '',
+            filteredItems: routes
+        };
+    }
 
-    const navItems = sections.sort(sortBy('sortOrder', 'name')).map(section => {
-        if (!groupedRoutes[section.name]) {
-            return;
-        }
+    onChangeHandler = (event) => {
+        let searchResults = routes.filter((navItem) => {
+            return navItem.sortOrder === 1 ? navItem : navItem.name.toLowerCase().includes(event.target.value.toLowerCase());
+        });
+        this.setState({
+            query: event.target.value,
+            filteredItems: [...searchResults]
+        });
+    };
 
-        sectionRoutes = groupedRoutes[section.name].sort(sortBy('sortOrder', 'name'));
-        return (
-            <ul className='frDocs-Nav__list' key={section.name}>
-                <li className='frDocs-Nav__headers'>{section.name}</li>
-                {sectionRoutes.map(route => (
-                    <li key={route.name}>
-                        <NavLink
-                            activeClassName='frDocs-Nav__item--active'
-                            className='frDocs-Nav__item'
-                            key={route.url}
-                            to={{ pathname: route.url }}>
-                            {route.name}
-                        </NavLink>
-                    </li>
-                ))}
+    render() {
+        let sectionRoutes;
+        const groupedRoutes = groupArray(this.state.filteredItems, 'section');
+
+        let navItems = sections.sort(sortBy('sortOrder', 'name')).map(section => {
+            if (!groupedRoutes[section.name]) {
+                return;
+            }
+
+            sectionRoutes = groupedRoutes[section.name].sort(sortBy('sortOrder', 'name'));
+            return (
+                <ul className='frDocs-Nav__list' key={section.name}>
+                    <li className='frDocs-Nav__headers'>{section.name}</li>
+                    {sectionRoutes.map(route => (
+                        <li key={route.name}>
+                            <NavLink
+                                activeClassName='frDocs-Nav__item--active'
+                                className='frDocs-Nav__item'
+                                key={route.url}
+                                to={{ pathname: route.url }}>
+                                {route.name}
+                            </NavLink>
+                        </li>
+                    ))}
+                </ul>
+            );
+        });
+
+        let navHomeSegment = navItems[0];
```

Comment:
It might be worth filtering out the `Home` object when you create `navItems`. This would allow you to avoid hard coding `0` a location on an array and having to use splice. 

You could then call the home route and render it manually using the render function. I think `routes.find({name: 'home'})` would return just the home object as well.